docs/projects/2024/architecture-transition/rpc.md

```diff
@@ -128,76 +77,136 @@ The move to JSON-RPC opens up the possibility for the front end to consolidate m
-| `/api/ui/v0/queries/{queryId}/shares/`                      | GET         |           |
-| `/api/ui/v0/queries/{queryId}/shares/`                      | POST        |           |
-| `/api/ui/v0/reflect/`                                       | POST        | _(none)_  |
-| `/api/ui/v0/schema_roles/{roleId}/`                         | DELETE      | _(none)_  |
-| `/api/ui/v0/schema_roles/`                                  | POST        | _(none)_  |
-| `/api/ui/v0/tables/{tableId}/records/delete/`               | DELETE      |           |
-| `/api/ui/v0/tables/{tableId}/shares/{shareId}/`             | PATCH       |           |
-| `/api/ui/v0/tables/{tableId}/shares/{shareId}/regenerate/`  | POST        |           |
-| `/api/ui/v0/tables/{tableId}/shares/`                       | GET         |           |
-| `/api/ui/v0/tables/{tableId}/shares/`                       | POST        |           |
-| `/api/ui/v0/users/{userId}/`                                | DELETE      |           |
-| `/api/ui/v0/users/{userId}/`                                | GET         |           |
-| `/api/ui/v0/users/{userId}/`                                | PATCH       |           |
-| `/api/ui/v0/users/{userId}/password_reset/`                 | POST        |           |
-| `/api/ui/v0/users/`                                         | GET         |           |
-| `/api/ui/v0/users/`                                         | POST        |           |
-| `/api/ui/v0/users/password_change/`                         | POST        |           |
+| Endpoint                                                    | HTTP Method | Function                                |
+| --                                                          | --          | --                                      |
+| `/api/db/v0/connections/{connectionId}/`                    | DELETE      | `connections.delete`                    |
+| `/api/db/v0/connections/{connectionId}/`                    | PATCH       | `connections.patch`                     |
+| `/api/db/v0/connections/`                                   | GET         | `connections.list`                      |
+| `/api/db/v0/data_files/{id}`                                | GET         | _(keep)_                                |
+| `/api/db/v0/data_files/{id}`                                | PATCH       | _(keep)_                                |
+| `/api/db/v0/data_files/`                                    | POST        | _(keep)_                                |
+| `/api/db/v0/links/`                                         | POST        | `data_modeling.add_link`                |
+| `/api/db/v0/queries/{queryId}/`                             | DELETE      | `explorations.delete`                   |
+| `/api/db/v0/queries/{queryId}/`                             | GET         | `explorations.get`                      |
+| `/api/db/v0/queries/{queryId}/`                             | PUT         | `explorations.replace`                  |
```

Comment:
Is this actually a thing? I.e., is there a UI flow for wholesale replacing an exploration?

Reply:
Yes. When the user clicks the "Save" button on an exploration, we go [here](https://github.com/mathesar-foundation/mathesar/blob/develop/mathesar_ui/src/systems/data-explorer/exploration-inspector/ExplorationTab.svelte#L76) and then [here](https://github.com/mathesar-foundation/mathesar/blob/develop/mathesar_ui/src/stores/queries.ts#L224). That's a PUT request. This is code that @pavish wrote, I think with Dom's help on the backend. I imagine they both had a good reasons for using PUT instead of PATCH, but I don't understand enough about the inner-workings of the exploration data structures to understand why. In my audit of REST endpoints, this one did stand out to me as the only endpoint where we're using a PUT request.

@pavish if you want to shed any light on your reasoning here, that might be helpful.